src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health = 100;
+  hidden = false;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Carnivore) && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        delete Animal.alive[Animal.alive.indexOf(animal)];
```

Comment:
Using `delete` on an array leaves holes in the array. This can lead to incorrect indexing. It's more efficient to use the `splice` method to remove elements from an array.